src/main/java/uk/gov/hmcts/reform/sendletter/config/FtpConfigProperties.java

```diff
@@ -102,7 +102,11 @@ public String getPrivateKey() {
     }
 
     public void setPrivateKey(String privateKey) {
-        this.privateKey = privateKey;
+        String wrapper = "-----";
```

Comment:
Not sure why we need to do this now and all of sudden private key is invalid. Could we update the private key instead of this ?

Reply:
spaces is the case now, actually

Reply:
is this because private key is loaded from env vars?

Reply:
if you check vault it is stored as a one liner. for microsoft cloud platform it's fine. for other platforms this fails to load and app crashes before starting. same experienced with ftp ui clients

Reply:
it would be nice to add a comment (in code) why this is being done imho.

Reply:
we can load the value of secret from az cli something like 
``` az keyvault secret set --name rootca --vault-name mykeyvault --file /path/to/file/privatekey.der```
which would result in below
```
{
  "attributes": {
    "created": "2018-01-04T15:43:18+00:00",
    "enabled": true,
    "expires": null,
    "notBefore": null,
    "recoveryLevel": "Purgeable",
    "updated": "2018-01-04T15:43:18+00:00"
  },
  "contentType": null,
  "id": "https://mykeyvault.vault.azure.net/secrets/rootca/68b70f7caa074f54b10d7dba2ff52e9e",
  "kid": null,
  "managed": null,
  "tags": {
    "file-encoding": "utf-8"
  },
  "value": "-----BEGIN CERTIFICATE-----\nMIIFADCCAuigAwIBAgIQS0Dm+q1kmqxBMcvB49AJlTANBgkqhkiG9w0BAQsFADAaMRgwFgYDVQQD\nDA9Eb2NrZXIgVExTIFJvb3QwHhcNMTgwMTA0MDExMjE3WhcNMzgwMTA0MDEyMjE2WjAaMRgwFgYD\nVQQDDA9Eb2NrZXIgVExTIFJvb3QwggIiMA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQDqxthQ\nZPxJY821/07C2W9d4Mg1mRf2lQmLhIbApGKsHa07rS4BusZ5qKrcP4I0xMY7wNFFyJIYC8oesKTC\nWZf9OhttIxCWA04RXkz0ISKMSAspUR1YBFsvAdHFVvt348y3aR1hmO6OXyqYlR3yms5A8hF8QpOQ\n+/hSaQ4MlfRwU8W6PNfV6SEzv7fGu0ZmZc+Wi7pHLrzIeLvGwkoEXJIQupe22h5GqkirZ501RvBB\n4SQkG5SHaKMTf5FpYr52C5B9bw7fh0KFXCDER6BD3sZCdv2Fz90yp37fwBBz60DhTrAcXetu6JSn\nFaB6GqtIg1G4ytQJnzcaOgq2YHeRaIuqzy5HuioQ9Dz32j4vUwNFD0rgF2/NaZAeVVK30XggB1+7\n+V2KpUPi2z9LZ/b5MktmdYdWLw+ShGSs4mUwESvWWPWo2ljAKm+01LmnwKEquVI/oXGLMiTFW6QP\nRLqKtTMHsJsEZsFueLGxoDaBKXTGvIYtjSQa9u3Wt1o2Ebaq+aNRDAHpoxU/4EEC4SotMyr2CFMW\nTxGDf589JBs8K02dJhfvFiBIiuAwpSS7+C3qGfnQKBXMGdyrZTrJfSwaSLKPy8VmWvChZF5l3oLF\nDR+I1G5fahDIFP1jcJJKIfX7sUvaouCLAZioR9t/q+BqzLmdxsslWIe4+vnQeSX55NA8EQIDAQAB\no0IwQDAOBgNVHQ8BAf8EBAMCAQYwDwYDVR0TAQH/BAUwAwEB/zAdBgNVHQ4EFgQU9AGt3fSrJxjZ\no44YJklXcyMv5pYwDQYJKoZIhvcNAQELBQADggIBAKFJzSBtwpBaRzXfuT0OLBz5tX9VHppuc8H7\nPmiv1NELAFa5Nwi0G/eQptB3etJzu8+gnH7SZoPaZjnqbLYgk4T1o1xrSKEwGJMqjD7PLsJqo8O1\n6fx6A8opH0hmh0qlvH3nSegjq/bmP8gKQTPHoCyAsbRGekUj7KdXqVsGYIClRVJEU49UK78a4aiI\nN/x0KMMY+dL8aZA3dKpqSqpkS/DWlNewiKe37/og2+pqYDZnCcd3N7IWThY+iFnuFQbOvsK+ipje\ndyzXt9dL3NSx/ZHT1L54ttL7LYfLKyk5yV63p+DONiVqppSnEZRMqXfJbWMFPjL73vEQpXFiq9CH\ntKTfmD+6YxjWuZ4BklF2Dry/2F7jXCd9HXp1wrm/sPg1Lr4SfU39kN1vZkVwWJGFOYXCInd7tpPn\nTjw6X2SK7VDaSH4WJAuSKtPzFSxoK1LWdLedv2OWkf/h3vJCQ3r3yBvHCoX4ghm6n8uYeXeDLjhI\np8KutuHENIr5ih+oq01mx9VEOaTVDDpiowB87LICFzyfuHVSB/nOZ+QoxCxXE7fZB+YpqP8aCyLb\nN7MCeVnOnAGzNlDiom1qgJxZIR/WUyXfQNEh4rLjI9w72rxw0aQG27+pM/4EYV3uem8J+i3OVTCL\n5BdmEo40rzK4JAYNsMUdgVlfDYbuGJOkfAJeDmq7\n-----END CERTIFICATE-----\n"
}
```

Reply:
then it is stored in application settings with spaces. my bad mentioning vault as i was actually checking app service

Reply:
I think it gets copied from azure key vault to app settings as an env var. In worst case this doesn't work then could we please add a comment on why we have done this code ?

Reply:
yes, i've included the comment. this pr is separation from aks enabling. and(!) is a result of a successful aks pipeline achieved step by step eliminating one bug after another. i'm not inventing anything here - this is a valid work-around since value which comes into configuration properties is not well formatted and crashes the app